src/Polly.Core/CircuitBreaker/Controller/CircuitStateController.cs

```diff
@@ -0,0 +1,343 @@
+using Polly.Strategy;
+
+namespace Polly.CircuitBreaker;
+
+/// <summary>
+/// Thread-safe controller that holds and manages the circuit breaker state transitions.
+/// </summary>
+internal sealed class CircuitStateController : IDisposable
+{
+    private readonly object _lock = new();
+    private readonly ScheduledTaskExecutor _executor = new();
+    private readonly OutcomeEvent<OnCircuitOpenedArguments>.Handler? _onOpened;
+    private readonly OutcomeEvent<OnCircuitClosedArguments>.Handler? _onClosed;
+    private readonly Func<OnCircuitHalfOpenedArguments, ValueTask>? _onHalfOpen;
+    private readonly TimeProvider _timeProvider;
+    private readonly ResilienceStrategyTelemetry _telemetry;
+    private readonly CircuitBehavior _behavior;
+    private readonly TimeSpan _breakDuration;
+    private long _blockedTill;
```

Comment:
`Until` rather than `Till`?

Reply:
changed to `DateTimeOffset? blockedUntil`